### PR TITLE
Added haus-bus.de adapter

### PR DESCRIPTION
**Pull request by hausbus**
This adapter provides ioBroker support for all haus-bus.de components. It was test for 2 years by many different customers.

### CODE REVIEW

**Comment by Apollon77** [comment]
Thank you @hausbus ,

please also just  the icon in this PR to link to the https://raw.githubusercontent.com domain like the other link. Thank you

**Comment by hausbus** [comment]
Hi Ingo,
was muss ich da genau machen?

Danke!
Herm
________________________________

**Comment by Apollon77** [comment]
Diese Zeile hier https://github.com/ioBroker/ioBroker.repositories/pull/1894/files#diff-1384831e45cb5b5fc71b065c642c35169d0ee196221e07b4b81075a0d3b0aed8R674 korrigieren? :-)

**Comment by hausbus** [comment]
fixed icon link

**Comment by GermanBluefox** [comment]
RE-CHECK!

**Comment by GermanBluefox** [comment]
RE-CHECK!

**Comment by GermanBluefox** [comment]
## Automated adapter checker

### [ioBroker.hausbus_de](https://github.com/hausbus/ioBroker.hausbus_de)
[Downloads](https://www.npmjs.com/package/iobroker.hausbus_de) 
[NPM](https://nodei.co/npm/iobroker.hausbus_de/)

:thumbsup: No errors found
- [ ] :eyes: [W171] "common.title" is deprecated in [io-package.json](https://github.com/hausbus/ioBroker.hausbus_de/blob/master/io-package.json)
- [ ] :eyes: [W156] Adapter should support admin 5 UI (jsonConfig)
- [ ] :eyes: [W400] Cannot find "hausbus_de" in latest repository
- [ ] :eyes: [W505] setTimeout found in "main.js", but no clearTimeout detected


*Add comment "RE-CHECK!" to start check anew*

**Comment by GermanBluefox** [comment]
## Automated adapter checker

### [ioBroker.hausbus_de](https://github.com/hausbus/ioBroker.hausbus_de)
[Downloads](https://www.npmjs.com/package/iobroker.hausbus_de) 
[NPM](https://nodei.co/npm/iobroker.hausbus_de/)

:thumbsup: No errors found
- [ ] :eyes: [W171] "common.title" is deprecated in [io-package.json](https://github.com/hausbus/ioBroker.hausbus_de/blob/master/io-package.json)
- [ ] :eyes: [W156] Adapter should support admin 5 UI (jsonConfig)
- [ ] :eyes: [W400] Cannot find "hausbus_de" in latest repository
- [ ] :eyes: [W505] setTimeout found in "main.js", but no clearTimeout detected


*Add comment "RE-CHECK!" to start check anew*

**Comment by Apollon77** [comment]
@hausbus Did you had a look at the comments of the automatic checker? Especially W156 and W505

**Comment by Apollon77** [comment]
Hello, thank for your submission, here some review comments:
* you use the release script, but did not updated to v3 correctly. There is another dependency for the iobroker plugin and a release config file needed. Please check release-script changelog for 3.0. It will not work as it is not because the io-package is not handled correctly.
* please add all config parameter in io-package in native are with their default values. If really no config variables are needed please remove the two dummy ones :-) (really not e.g. the network interface to use for devices with multiple or such?) 
* your config states that compact mode is supported, but code is not written for it. https://github.com/hausbus/ioBroker.hausbus_de/blob/main/io-package.json#L74 .Would be cool to see it supported ... 
* see https://github.com/hausbus/ioBroker.hausbus_de/commit/4c331b3930457a9b8d8dbb42d06c7349644d61e8#r84160575

Thank you for checking and adjusting to the comments

Ingo

**Comment by hausbus** [comment]
Hey Ingo,
 thank you for your hints.
How comes that the release Skript ist not correct although I used the latest adapter creator?

Hermann
________________________________

**Comment by Apollon77** [comment]
Hm ... did the creater created it would the plugin in 3.x version? OR was the creation creating it as 2.x version of the release script nd you upgraded to 3?

If it was created as 3.x this is worh an creator issue!

**Comment by Apollon77** [comment]
Pleas ejust poke when review comments are adressed

**Comment by hausbus** [comment]
Wie poked man denn ?

Von: Ingo Fischer ***@***.***>
Gesendet: Sonntag, 18. September 2022 17:11
An: ioBroker/ioBroker.repositories ***@***.***>
Cc: Hermann Hoeschen ***@***.***>; Mention ***@***.***>
Betreff: Re: [ioBroker/ioBroker.repositories] Added haus-bus.de adapter (PR #1894)


Pleas ejust poke when review comments are adressed

**Comment by Apollon77** [comment]
indem Du einfach hier kommentierst und Feedback gibst :-)

**Comment by hausbus** [comment]
Ich glaube ich muss mich noch etwas an diese Arbeitsweise gewöhnen. Ich bin den github style noch nicht so gewöhnt.
Waren jetzt noch weitere Dinge offen? Die dependencies hatte ich gerade entfernt

**Comment by Apollon77** [comment]
Ich hatte in https://github.com/ioBroker/ioBroker.repositories/pull/1894#issuecomment-1249265984 alles zusammengefasst ... da alles angeschaut?

**Comment by hausbus** [comment]
Yop, das sollte alles erledigt sein

**Comment by Apollon77** [comment]
ok, then here my recheck:

You removed the dependencies os/dgram. good. Please also release a new version now :-)

To jsut set compact to false in the settings was the cheapest of all solutions ... formally we require all new adapters to have compact mode support. DO you think you can adjust the code? It should not be that much:
* add code like this to the end https://github.com/Apollon77/ioBroker.nut/blob/master/nut.js#L469-L475
* move the "adapter initialization in a startAdapter method that returns the adapter instance afterwards (see https://github.com/Apollon77/ioBroker.nut/blob/master/nut.js#L25-L30) ...
Thats basically it ... and do a short testing by enabling compact mode support for the js.controller you use and enable compact mode for the adater. Would be awesome. Thank you!

**Comment by hausbus** [comment]
poke :-)
I added the code for the compact mode. It seems to work. At least it still started. I don't think that I can test any further right?
Or is it possible to verify that the adapter is now running in the js-controller instance ?
New release is 1.0.17

**Comment by Apollon77** [comment]
Sure you can test compact mode and it would be cool if you do :-) See https://github.com/ioBroker/ioBroker.js-controller#start-adapter-instances-in-compact-mode ... beside CLI you can also do that via Admin

The release script increases both versions WHEN oyu use v2.x OR do a correct upgrade to 3.x of the rerlease script.This is what I wrote above! Version 3 of the release script is a major change ... Major means Breaking! see https://github.com/AlCalzone/release-script#plugins ... So you need another plugin dependency and a release script config. Then this works again.

For the Auto-Merge support you need to add a Github action Secret, see https://github.com/hausbus/ioBroker.hausbus_de/blob/main/.github/workflows/dependabot-auto-merge.yml#L22

**Comment by hausbus** [comment]
I now testet compact mode. After moving the adapter to compact group 0 the admin shows mit that my adapter is no longer consuming any memory :)
I already tried to upgrate to release script 3.x like you recommended but missed adding the addon to the command line call. 

Now my build is still failing due to an invalid lock file:
npm ERR! Invalid: lock file's @alcalzone/release-script@2.2.2 does not satisfy @alcalzone/release-script@^3.5.9

Any Idea how to fix that ?

Except that you could now check version 1.0.20. Thanks!

Herm

**Comment by Apollon77** [comment]
>  npm ERR! Invalid: lock file's @alcalzone/release-script@2.2.2 does not satisfy @alcalzone/release-script@^3.5.9

Seems your package-lock run out if sync. you need to run "npm i "in the adapter dir on your dev system when updating dependencies, then package-locj is also updated

**Comment by hausbus** [comment]
Ok, that seems to be fixed now. Thanks

**Review by Apollon77 — CHANGES_REQUESTED** [review]
Hey, so reveiw wise we are done and I would merge, but for these two topics ... please adjust or htell me what version and we can merge it!

File: sources-dist-stable.json [hunk 1]
```diff
@@ -716,6 +716,12 @@
     "type": "iot-systems",
     "version": "1.3.0"
   },
+  "hausbus_de": {
+    "meta": "https://raw.githubusercontent.com/hausbus/ioBroker.hausbus_de/main/io-package.json",
+    "icon": "https://github.com/hausbus/ioBroker.hausbus_de/blob/main/admin/hausbusde.png",
+    "type": "iot-systems",
+    "version": "1.0.16"
```
[the remark this answers is not in the document]
Comment (hausbus) [reply sources-dist-stable.json:723]: Hi Apollon,
sorry for my lack of experience but how do I get the file to raw.githubusercontent.com? If I just change the url the file cannot be found.

[the remark this answers is not in the document]
Comment (Apollon77) [reply sources-dist-stable.json:723]: https://raw.githubusercontent.com/hausbus/ioBroker.hausbus_de/main/admin/hausbusde.png should be correct ....  the structure is slightly different ... but there should be >500 comparable entries to look at *gg*

[the remark this answers is not in the document]
Comment (hausbus) [reply sources-dist-stable.json]: How does it look now? I also updated the version to the latest release 1.1.2

[the remark this answers is not in the document]
Comment (Apollon77) [reply sources-dist-stable.json]: Innfact yopi set the version there which is "proofed stable, some days released already from your perspecticve and indeally also no issues reported and such"

[the remark this answers is not in the document]
Comment (hausbus) [reply sources-dist-stable.json]: Ähm... What does that mean ?

[the remark this answers is not in the document]
Comment (Apollon77) [reply sources-dist-stable.json]: Ok I switch to german:
DIe Idee des "Stable" Repositories und damit der Versionsangabe ist es als Stabil bewiesene Versionen für die Allgemeinheit der User anzubieten. neue Versionen kommen nach Publish auf npm automatisch ins Beta Repo. Wenn dann eine Version dort 1-2 Wochen stabil lieff, gutes Feedback und keine Fehler vorliegen dann machst Du einen PR mit einer neuen Stabilen Version.

Also die Stable Version ist die wo Du sagst "Die ist stabil und gut für alle"

[the remark this answers is not in the document]
Comment (hausbus) [reply sources-dist-stable.json]: Ok, verstanden. Dann erstelle ich in 2 Wochen einen neuen Pullrequest. Was passiert mit diesem hier? Wird der geschlossen ?

**Comment by Apollon77** [comment]
Welcome in the repo :-)

**Comment by hausbus** [comment]
Hallo Ingo,
kannst du mir nochmal auf die Sprünge helfen?
Ich wollte gerade unseren Adapter aktualisieren und bekomme nun folgendes:

npm ERR! notsup Unsupported engine for ***@***.***: wanted: {"node":">=14.18.0 < 15 || >= 16.4.0"} (current: {"node":"12.22.12","npm":"6.14.16"})

Muss ich irgendwo neues Dependencies eintragen ?

Danke!
Herm

Von: Ingo Fischer ***@***.***>
Gesendet: Mittwoch, 16. November 2022 09:35
An: ioBroker/ioBroker.repositories ***@***.***>
Cc: Hermann Hoeschen ***@***.***>; Mention ***@***.***>
Betreff: Re: [ioBroker/ioBroker.repositories] Added haus-bus.de adapter (PR #1894)


Merged #1894<https://github.com/ioBroker/ioBroker.repositories/pull/1894> into master.

**Comment by mcm1957** [comment]
Dieser PR ist wohl nicht der geeignete Platz für Nachfragen :-)
Bitte Telegram oder Forum benutzen.
Das Problem scheint zu sein, dass auf dem System node12 läuft. Dieses wird nicht mehr unterstützt.

**Comment by Apollon77** [comment]
@hausbus Wo beskommst du das? Bei den tests? Dannbitte nodejs 12 aus den test rausnehmen! Siehe https://forum.iobroker.net/topic/63825/controller-v5-kein-support-von-node-js-12

**Comment by hausbus** [comment]
Vielen Dank, hat geklappt.
Nochmal eine Frage. Ich habe gerade eine neue Version released. Wie lange dauert es, bis die Version im Betarepository angezeigt wird?
Oder passiert das nicht automatisch, wenn ich die Version von 1.1.5 auf 1.2.1 erhöht habe ?

Danke!
Herm


Von: Ingo Fischer ***@***.***>
Gesendet: Sonntag, 26. März 2023 11:36
An: ioBroker/ioBroker.repositories ***@***.***>
Cc: Hermann Hoeschen ***@***.***>; Mention ***@***.***>
Betreff: Re: [ioBroker/ioBroker.repositories] Added haus-bus.de adapter (PR #1894)


@hausbus<https://github.com/hausbus> Wo beskommst du das? Bei den tests? Dannbitte nodejs 12 aus den test rausnehmen! Siehe https://forum.iobroker.net/topic/63825/controller-v5-kein-support-von-node-js-12

**Comment by mcm1957** [comment]
Latest wurd typich 2x Rag ca 14:00 und 2:00 aktualisiert. Latest update sillte immer automatisch erfolgenMcM
-------- Ursprüngliche Nachricht --------Von: Hermann Hoeschen ***@***.***> Datum: 26.03.23  13:07  (GMT+01:00) An: "ioBroker/ioBroker.repositories" ***@***.***> Cc: "Martin M." ***@***.***>, Comment ***@***.***> Betreff: Re: [ioBroker/ioBroker.repositories] Added haus-bus.de adapter (PR #1894) 
Vielen Dank, hat geklappt.

Nochmal eine Frage. Ich habe gerade eine neue Version released. Wie lange dauert es, bis die Version im Betarepository angezeigt wird?

Oder passiert das nicht automatisch, wenn ich die Version von 1.1.5 auf 1.2.1 erhöht habe ?



Danke!

Herm





Von: Ingo Fischer ***@***.***>

Gesendet: Sonntag, 26. März 2023 11:36

An: ioBroker/ioBroker.repositories ***@***.***>

Cc: Hermann Hoeschen ***@***.***>; Mention ***@***.***>

Betreff: Re: [ioBroker/ioBroker.repositories] Added haus-bus.de adapter (PR #1894)





@hausbus<https://github.com/hausbus> Wo beskommst du das? Bei den tests? Dannbitte nodejs 12 aus den test rausnehmen! Siehe https://forum.iobroker.net/topic/63825/controller-v5-kein-support-von-node-js-12

**Pull request merged**